src/canmatrix/formats/arxml.py

```diff
@@ -212,7 +212,8 @@ def get_child(self, parent, tag_name):
             reference = self.find(tag_name + '-REF', parent)
             if reference is not None:
                 ret = self.get_short_name_path(reference.text)
-                raise "use follow_ref!"
```

Comment:
Hi @leixao,
this "raise" should stay here because it only raises if used the "wrong" (old) way.

Could you provide an example, where this exception raises?
Maybe something like a call-tree?

Reply:
hi @ebroecker
The following is the call stack when converting my arxml file to dbc:
Traceback (most recent call last):
  File "C:\Python\Python39\Scripts\canconvert-script.py", line 33, in <module>
    sys.exit(load_entry_point('canmatrix===0caf5fe.post49-ga798bfc', 'console_scripts', 'canconvert')())
  File "C:\Python\Python39\lib\site-packages\click-8.0.0a1-py3.9.egg\click\core.py", line 1025, in __call__
    return self.main(*args, **kwargs)
  File "C:\Python\Python39\lib\site-packages\click-8.0.0a1-py3.9.egg\click\core.py", line 955, in main
    rv = self.invoke(ctx)
  File "C:\Python\Python39\lib\site-packages\click-8.0.0a1-py3.9.egg\click\core.py", line 1279, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Python\Python39\lib\site-packages\click-8.0.0a1-py3.9.egg\click\core.py", line 710, in invoke
    return callback(*args, **kwargs)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\cli\convert.py", line 135, in cli_convert
    canmatrix.convert.convert(infile, outfile, **options)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\convert.py", line 40, in convert
    dbs = canmatrix.formats.loadp(infile, **options)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\__init__.py", line 70, in loadp
    return load(fileObject, import_type, key, **options)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\__init__.py", line 87, in load
    dbs = module_instance.load(file_object, **options)  # type: ignore
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\arxml.py", line 1947, in load
    result.update(decode_can_helper(ea, float_factory, ignore_cluster_info))
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\arxml.py", line 1842, in decode_can_helper
    frame = get_frame(frameTrig, ea, multiplex_translation, float_factory)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\arxml.py", line 1513, in get_frame
    get_signals(pdu_sig_mapping, new_frame, ea, None, float_factory)
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\arxml.py", line 1155, in get_signals
    base_type = ea.get_child(datdefprops, "BASE-TYPE")
  File "C:\Python\Python39\lib\site-packages\canmatrix-0caf5fe.post49_ga798bfc-py3.9.egg\canmatrix\formats\arxml.py", line 215, in get_child
    raise "use follow_ref!"
TypeError: exceptions must derive from BaseException

It seems that the direct element is not found and the reference is searched, but why is an exception thrown when a reference is not none? Is there a problem with my arxml file?

Reply:
Hi @leixao

The reason for this exception is exactly what you describe. 
I added it to remove this "magic" behaviour from the arxml-parser. 
This is very disturbing for users who read the code. Thus I want "follow-ref" if its a "-REF" element and get_child if it's not a "-REF"-element.

Thanks for posting the callstac of the error, thus I am able to fix it.